src/enzyme_ad/jax/Passes/LLVMToAffineAccess.cpp

```diff
@@ -466,6 +466,40 @@ struct AffineExprBuilder {
     return failure();
   }
 
+  static bool isStrictlyAffine(AffineExpr expr) {
+
+    if (AffineBinaryOpExpr bOpExpr = dyn_cast<AffineBinaryOpExpr>(expr)) {
+      bool ret_flag = false;
+      switch (bOpExpr.getKind()) {
+      case mlir::AffineExprKind::Add:
+        // do nothing
+        break;
+      case mlir::AffineExprKind::Mul: {
+        bool lhsConst = isa<AffineConstantExpr>(bOpExpr.getLHS());
+        bool rhsConst = isa<AffineConstantExpr>(bOpExpr.getRHS());
+
+        // either LHS or RHS has to be a constant
+        if (!lhsConst && !rhsConst)
```

Comment:
the logic here needs to be more complex to be correct

for example, will accidentally fail if you have, say (s1 + 1) * (s2 + 2), since the lhs/rhs aren't symbols / constants

in any case, we can revisit this PR post GB